packages/flutter_reactive_ble/lib/src/reactive_ble.dart

```diff
@@ -154,24 +152,28 @@ class FlutterReactiveBle {
   /// Be aware that a read request could be satisfied by a notification delivered
   /// for the same characteristic via [characteristicValueStream] before the actual
   /// read response arrives (due to the design of iOS BLE API).
-  Future<List<int>> readCharacteristic(
-      QualifiedCharacteristic characteristic) async {
+  ///
+  /// This method assumes there is a single characteristic with the ids specified in [characteristic]. If there are
+  /// multiple characteristics with the same id on a device, use [resolve] to find them all. Or use
+  /// [getDiscoveredServices] to select the [Service]s and [Characteristic]s you're interested in.
+  Future<List<int>> readCharacteristic(QualifiedCharacteristic characteristic) async {
```

Comment:
I see no compelling reason to deprecate them. It would force the majority of users who deal with devies without multiple services/chars of the same type to refactor their code.